lib/scenic/statements.rb

```diff
@@ -82,17 +80,10 @@ def update_view(name, version: nil, revert_to_version: nil, materialized: false)
         raise ArgumentError, "version is required"
       end
 
-      if materialized
-        raise ArgumentError, "Updating materialized views is not supported "\
-          "because it would cause any indexes to be dropped. Please use "\
-          "'drop_view' followed by 'create_view', being sure to also recreate "\
-          "any previously-existing indexes."
+      Scenic.database.reapplying_indexes(on: name) do
+        drop_view(name, revert_to_version: revert_to_version, materialized: materialized)
```

Comment:
I'm also fine with this. I feel like readability is better with the longer line.